### PR TITLE
fix default load balancer with warmup duration

### PR DESCRIPTION
**Pull request by ramaraochavali**
Fixes https://github.com/istio/istio/issues/39430

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 1]
```diff
@@ -725,12 +719,22 @@ func applyLoadBalancer(c *cluster.Cluster, lb *networking.LoadBalancerSettings,
...
 	default:
-		c.LbPolicy = defaultLBAlgorithm()
+		ApplyDefaultLoadBalancer(c, lb)
 	}
 
 	ApplyRingHashLoadBalancer(c, lb)
 }
 
+// ApplyDefaultLoadBalancer will set the DefaultLBPolicy and create an LbConfig if used in LoadBalancerSettings
+func ApplyDefaultLoadBalancer(c *cluster.Cluster, loadbalancer *networking.LoadBalancerSettings) {
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/cluster.go:729]: prefer calling` applyDefaultLoadBalancerAlgorithm`. ApplyDefaultLoadBalancer is too wide

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 2]
```diff
@@ -725,12 +719,22 @@ func applyLoadBalancer(c *cluster.Cluster, lb *networking.LoadBalancerSettings,
 		c.LbPolicy = cluster.Cluster_CLUSTER_PROVIDED
 		c.ClusterDiscoveryType = &cluster.Cluster_Type{Type: cluster.Cluster_ORIGINAL_DST}
 	default:
-		c.LbPolicy = defaultLBAlgorithm()
+		ApplyDefaultLoadBalancer(c, lb)
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/cluster.go:722]: c.LbPolicy is unset?

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/cluster.go]: oh. Sorry. I thought I pushed that commit, but did not. Now pushed.

File: pilot/pkg/networking/core/v1alpha3/cluster.go (hunk 1 above)
Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/cluster.go]: I renamed to applyDefaultSimpleLoadBalancer as that func is intended to be used to set other lb settings in future if they come up - it is not just lb algo.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**